cpp/include/cudf/utilities/error.hpp

```diff
@@ -77,37 +77,59 @@ struct fatal_cuda_error : public cuda_error {
  * @brief Macro for checking (pre-)conditions that throws an exception when
  * a condition is violated.
  *
+ * Defaults to throwing `cudf::logic_error`, but a custom exception may also be
+ * specified.
+ *
  * Example usage:
  *
  * @code
- * CUDF_EXPECTS(lhs->dtype == rhs->dtype, "Column type mismatch");
+ * // throws cudf::logic_error
+ * CUDF_EXPECTS(lhs.type() != rhs.type(), "Column type mismatch");
```

Comment:
`==` ? (or else this is expecting lhs and rhs to having mismatching types so the error string is backwards?)